sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -3441,6 +3460,18 @@ def value_counts(self, subset=None, sort=False, normalize=False,
       else:
         return result
 
+  @frame_base.with_docs_from(pd.DataFrame)
+  @frame_base.args_to_kwargs(pd.DataFrame)
+  @frame_base.populate_defaults(pd.DataFrame)
+  def compare(self, other, **kwargs):
```

Comment:
Hi @TheNeuralBit, I have a similar issue here as in `idxmin` and `idxmax` proxy. I wasn't able to create a valid proxy since the expected proxy is a DataFrame with a MultiIndex and its structure depends on the differences between the inputs. So, I'm not sure how this proxy can be created beforehand. Any suggestion on how this can be solved?

Reply:
Hi @roger-mike, looking through the docs for this operation [here](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.compare.html), I think we will need to be restrictive about the arguments we support. As you point out, the default (`align_axis=1`, `keep_shape=False`) will drop columns that are equivalent. Since that makes the shape depend on the input data, we should raise `WontImplementError(reason="non-deferred-columns")` in that case.

We should still be able to support `align_axis=1`, `keep_shape=True` though. And I think we should be able to support `align_axis=0` no matter what the other args are.

Does that make sense?

Reply:
Thanks for your comments @TheNeuralBit. I handled the (`align_axis=1`, `keep_shape=False`) case as you suggest. The other cases work just fine. I also disabled proxy check for the (`align_axis=0`, `keep_shape=False`) test case since its result also depends on the input data. Let me know what you think 👍 .

Reply:
Great, thanks!

One more thing: any time there's a caveat like the wontimplementerror here, we should add a docstring for it, we have infrastructure that puts this in a "Differences from pandaas" section. See [here](https://beam.apache.org/releases/pydoc/2.33.0/apache_beam.dataframe.frames.html#module-apache_beam.dataframe.frames).

Something like:
```suggestion
  def compare(self, other, **kwargs):
    """The default values ``align_axis=1 and ``keep_shape=False`` are not supported,  because the output columns depend on the data. To use ``align_axis=1``, please specify ``keep_shape=True``."""
```